*: audit all duration settings to ensure they have validation

yuzefovich opened the pull request:
This PR contains a couple of commits that audit all duration settings and ensure that they have validation functions. In most cases where there wasn't one, non-negative validation is added. The exceptions are:
- positive validation:
  - `kv.gc.txn_cleanup_threshold`
  - `jobs.scheduler.pace`
  - `sql.stats.max_timestamp_age`.
- non-positive validation:
  - `sql.crdb_internal.table_row_statistics.as_of_time`.

Non-negative validation is now assumed to be the default, so all its explicit usages are removed.

See each commit for details.

Epic: None
Release note: None

yuzefovich committed:
*: audit all duration settings to add validation function

This commit adjusts the DurationSetting so that if no validation
function is provided, it assumes the non-negative duration.

This commit audits all duration settings to add validation functions. In
vast majority cases non-negative duration seems reasonable, for three
I'm adding positive validation:
- `kv.gc.txn_cleanup_threshold`
- `jobs.scheduler.pace`
- `sql.stats.max_timestamp_age`.

`sql.crdb_internal.table_row_statistics.as_of_time` got a non-positive
validation.

Note that since we're adding new validation logic, it can be possible
for existing setting values to not pass the validation anymore. In such
case the default will be used instead (and a log message will be
written) which seems like a reasonable behavior.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/144028)
<!-- Reviewable:end -->

yuzefovich committed:
*: cleanup some duration settings

This commit should be almost a no-op and it does a few cleanup
things like replacing custom validation function with the helper
equivalent and putting the default value on a separate line.

The only non-equivalent change is introduction of non-negative
validation to `kv.dist_sender.circuit_breaker.cancellation.write_grace_period`
which seems reasonable.

Release note: None

yuzefovich commented:
@arulajmani @msbutler explicitly requested review from two of you since you were kind enough to respond to my questions on slack 😃

I intend to squash the second and the third commits together, but for now they are separate to highlight which settings get the non-negative validation.

msbutler reviewed (approved):
LGTM!

yuzefovich commented:
TFTR!

bors r+

yuzefovich commented:
bors r-

(forgot to squash)

yuzefovich commented:
bors r+
